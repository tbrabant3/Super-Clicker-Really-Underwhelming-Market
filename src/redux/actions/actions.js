export const INCREMENT_COUPON_COUNT = 'INCREMENT_COUPON_COUNT';
export const ADD_NEW_UPGRADE = 'ADD_NEW_UPGRADE';
export const REMOVE_UPGRADE = 'REMOVE_UPGRADE';
export const SUBTRACT_COUPON_AMOUNT = 'SUBTRACT_COUPON_AMOUNT';
export const ADD_COUPON_AMOUNT = 'ADD_COUPON_AMOUNT';
