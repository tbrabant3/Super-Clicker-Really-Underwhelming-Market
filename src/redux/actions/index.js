import { INCREMENT_COUPON_COUNT } from './actions';

export const incrementCouponCount = () => ({
	type: INCREMENT_COUPON_COUNT
});
