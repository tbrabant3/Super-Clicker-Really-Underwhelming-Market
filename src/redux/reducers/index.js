import { combineReducers } from 'redux';
import CouponsReducer from './coupons';
import UpgradesReducer from './upgrades';
import CPSReducer from './coupons-per-second';
import ThemeReducer from './header';

export default combineReducers({
	CouponsReducer,
	UpgradesReducer,
	CPSReducer,
	ThemeReducer
});
