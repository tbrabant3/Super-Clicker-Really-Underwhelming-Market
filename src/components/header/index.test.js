import React from 'react';
import reducer from '../../redux/reducers/header';
import * as actions from '../../redux/actions';
import * as types from '../../redux/actions/actions';

describe('Redux Testing', () => {
	const { open } = window;

	beforeAll(() => {
		delete window.open;
		window.open = jest.fn();
	});

	afterAll(() => {
		window.open = open;
	});

	it('should create a toggle light and dark theme action', () => {
		const expectedAction = {
			type: types.TOGGLE_LIGHT_DARK_THEME
		};
		expect(actions.toggleLightDarkTheme()).toEqual(expectedAction);
	});

	it('should handle TOGGLE_LIGHT_DARK_THEME', () => {
		expect(
			reducer([], {
				type: types.TOGGLE_LIGHT_DARK_THEME
			})
		).toEqual({
			theme: 'light'
		});
	})
});
