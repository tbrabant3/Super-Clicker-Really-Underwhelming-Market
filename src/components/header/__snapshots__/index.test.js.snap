// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Testing Should Render Header 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <h6
      className="MuiTypography-root MuiTypography-h6"
    >
      SUPER CLICKER REALLY UNDERWHELMING MARKET
    </h6>
    <span
      className="MuiSwitch-root"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-86 MuiSwitch-switchBase MuiSwitch-colorSecondary PrivateSwitchBase-checked-87 Mui-checked"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            checked={true}
            className="PrivateSwitchBase-input-89 MuiSwitch-input"
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="MuiSwitch-thumb"
          />
        </span>
      </span>
      <span
        className="MuiSwitch-track"
      />
    </span>
  </div>
</header>
`;
